### PR TITLE
Add ability to use as a flake

### LINKED ISSUE #44: Add support for using as a flake
Could use https://github.com/edolstra/flake-compat to support backward-compatibility.

### PR DESCRIPTION
**Pull request by jdahm**
Add a `flake.nix` that imports `default.nix`, and refactor out the overlay to `overlay.nix`.

Resolves #44.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -1,39 +1,13 @@
...
-  nixpkgs = import (builtins.fetchTarball {
-    # Descriptive name to make the store path easier to identify
-    name = "nixos-release-21.11";
-    url =
-      "https://github.com/nixos/nixpkgs/archive/573095944e7c1d58d30fc679c81af63668b54056.tar.gz";
-    # Hash obtained using `nix-prefetch-url --unpack <url>`
-    sha256 = "07s5cwhskqvy82b4rld9b14ljc0013pig23i3jx3l3f957rk95pg";
-  }) { overlays = [ overlay ]; };
-in nixpkgs
+{ system ? builtins.currentSystem, pkgs ? import (builtins.fetchTarball {
```
Comment (nbren12) [inline default.nix:2]: Can you update the readme to reflect this change? We'll also need to keep it in mind in downstream packags e.g. fv3gfs-fortran.

File: overlay.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
+let
```
Comment (nbren12) [inline overlay.nix:1]: It seems like this refactor is not 100% necessary, but I'm not opposed.

**Review by nbren12 — CHANGES_REQUESTED** [review]
Just requesting the readme change.

File: overlay.nix (hunk 2 above)
Comment (jdahm) [reply overlay.nix:1]: Definitely not necessary, but otherwise there were multiple levels of let-in, which became confusing to read.

**Comment by jdahm** [comment]
I wanted to add the pre-commit-hooks to the flake, but I ran into this issue: https://github.com/cachix/pre-commit-hooks.nix/issues/134. Will fix in a follow-up PR once that is addressed.

File: default.nix (hunk 1 above)
Comment (jdahm) [reply default.nix:2]: Done!

**Pull request merged**